src/bun.js/node/util/validators.zig

```diff
@@ -0,0 +1,233 @@
+const std = @import("std");
+const bun = @import("root").bun;
+const string = bun.string;
+const JSC = @import("root").bun.JSC;
+const JSValue = JSC.JSValue;
+const JSGlobalObject = JSC.JSGlobalObject;
+const ZigString = JSC.ZigString;
+
+pub fn getTypeName(globalObject: *JSGlobalObject, value: JSValue) ZigString {
+    var js_type = value.jsType();
+    if (js_type.isArray()) {
+        return ZigString.static("array").*;
+    }
+    return value.jsTypeString(globalObject).getZigString(globalObject);
+}
+
+pub fn throwErrInvalidArgValue(
+    globalThis: *JSGlobalObject,
+    comptime fmt: string,
+    args: anytype,
+) !void {
+    @setCold(true);
+    const err = JSC.toTypeError(JSC.Node.ErrorCode.ERR_INVALID_ARG_VALUE, fmt, args, globalThis);
+    globalThis.vm().throwError(globalThis, err);
+    return error.InvalidArgument;
+}
+
+pub fn throwErrInvalidArgTypeWithMessage(
+    globalThis: *JSGlobalObject,
+    comptime fmt: string,
+    args: anytype,
+) !void {
+    @setCold(true);
+    const err = JSC.toTypeError(JSC.Node.ErrorCode.ERR_INVALID_ARG_TYPE, fmt, args, globalThis);
+    globalThis.vm().throwError(globalThis, err);
+    return error.InvalidArgument;
+}
+
+pub fn throwErrInvalidArgType(
+    globalThis: *JSGlobalObject,
+    comptime name_fmt: string,
+    name_args: anytype,
+    comptime expected_type: []const u8,
+    value: JSValue,
+) !void {
+    @setCold(true);
+    const actual_type = getTypeName(globalThis, value);
+    try throwErrInvalidArgTypeWithMessage(globalThis, "\"" ++ name_fmt ++ "\" property must be of type {s}, got {s}", name_args ++ .{ expected_type, actual_type });
+}
+
+pub fn throwRangeError(
+    globalThis: *JSGlobalObject,
+    comptime fmt: string,
+    args: anytype,
+) !void {
+    @setCold(true);
+    const err = globalThis.createRangeErrorInstanceWithCode(JSC.Node.ErrorCode.ERR_OUT_OF_RANGE, fmt, args);
+    globalThis.vm().throwError(globalThis, err);
+    return error.InvalidArgument;
+}
+
+/// -(2^53 - 1)
+pub const NUMBER__MIN_SAFE_INTEGER: i64 = -9007199254740991;
+/// (2^53 – 1)
+pub const NUMBER__MAX_SAFE_INTEGER: i64 = 9007199254740991;
+
+pub fn validateInteger(globalThis: *JSGlobalObject, value: JSValue, comptime name_fmt: string, name_args: anytype, min_value: ?i64, max_value: ?i64) !void {
```

Comment:
i like this pattern. i wonder if making it return the validated data would be a nicer pattern.

Reply:
done!